src/Aspire.Hosting.Azure.WebPubSub/AzureWebPubSubExtensions.cs

```diff
@@ -64,4 +92,82 @@ public static IResourceBuilder<AzureWebPubSubResource> AddAzureWebPubSub(this ID
                       .WithParameter(AzureBicepResource.KnownParameters.PrincipalType)
                       .WithManifestPublishingCallback(resource.WriteToManifest);
     }
+
+    /// <summary>
+    /// Add hub settings
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="hubName">The hub name. Hub name is case-insensitive.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<AzureWebPubSubHubResource> AddHub(this IResourceBuilder<AzureWebPubSubResource> builder, [ResourceName] string hubName)
+    {
+        AzureWebPubSubHubResource? hubResource;
+        if (!builder.Resource.Hubs.TryGetValue(hubName, out hubResource))
+        {
+            hubResource = new AzureWebPubSubHubResource(hubName, builder.Resource);
+            builder.Resource.Hubs[hubName] = hubResource;
+        }
+        var hubBuilder = builder.ApplicationBuilder.CreateResourceBuilder(hubResource);
+        return hubBuilder;
+    }
+
+    /// <summary>
+    /// Add event handler setting with expression
+    /// </summary>
+    /// <param name="builder">The builder for a Web PubSub hub.</param>
+    /// <param name="urlTemplateExpression">The expression to evaluate the URL template configured for the event handler.</param>
+    /// <param name="userEventPattern">The user event pattern for the event handler.</param>
+    /// <param name="systemEvents">The system events for the event handler.</param>
+    /// <param name="authSettings">The auth settings configured for the event handler.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<AzureWebPubSubHubResource> AddEventHandler(this IResourceBuilder<AzureWebPubSubHubResource> builder,
```

Comment:
you'll want an overload that takes ReferenceExpression as well, that can be follow up